alacritty/src/display/mod.rs

```diff
@@ -411,12 +413,23 @@ impl Display {
         let metrics = glyph_cache.font_metrics();
         let (cell_width, cell_height) = compute_cell_size(config, &metrics);
 
+        // In macOS, for keeping the window position after resizing
+        // at the top left window corner, it must be visible.
+        #[cfg(target_os = "macos")]
+        window.set_visible(true);
+
         // Resize the window to account for the user configured size.
         if let Some(dimensions) = config.window.dimensions() {
             let size = window_size(config, dimensions, cell_width, cell_height, scale_factor);
             window.request_inner_size(size);
         }
 
+        // Set the window position to account for the user configured position.
+        #[cfg(target_os = "macos")]
+        if let Some(position) = config.window.position {
+            window.set_outer_position(PhysicalPosition::new(position.x as u32, position.y as u32));
+        }
```

Comment:
Did you just add this using trial and error? We're already setting the position on startup, why are you setting it again?

This does not seem like the proper solution, it just looks like we're moving it a bunch hoping that one of the operations will stick.